src/antsibull/sanity_tests.py

```diff
@@ -0,0 +1,361 @@
+# Copyright (C) 2023 Maxwell G <maxwell@gtmx.me>
+# SPDX-License-Identifier: GPL-3.0-or-later
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+"""
+Utilities to run sanity tests accross collections
+"""
+
+from __future__ import annotations
+
+import datetime
+import json
+import shlex
+import shutil
+import sys
+from collections.abc import Collection, Iterable, Iterator, Sequence
+from dataclasses import dataclass
+from functools import partial
+from pathlib import Path
+from subprocess import CompletedProcess
+from typing import TYPE_CHECKING, Any, TypedDict
+
+from antsibull_core import app_context
+from antsibull_core.logging import log
+from antsibull_core.subprocess_util import log_run
+from antsibull_core.yaml import store_yaml_file
+from packaging.version import Version
+
+from antsibull.constants import SANITY_TESTS_BANNED_IGNORES, SANITY_TESTS_DEFAULT
+from antsibull.types import CollectionName, add_dataclass_yaml_type
+
+if TYPE_CHECKING:
+    from _typeshed import StrPath
+
+mlog = log.fields(mod=__name__)
+
+
+class SanityOutput(TypedDict):
+    """
+    Mapping of `ansible-test sanity` output and other related data
+    """
+
+    cmd: list[str]
+    returncode: int
+    stdout: str
+    stderr: str
+    runtime: float
+    test_json: dict[str, Any]
+    ignore_entries: list[IgnoreEntry]
+    banned_ignore_entries: list[IgnoreEntry]
```

Comment:
`banned_ignore_entries` isn't used at the moment, right?

Reply:
Ah, I see, it's used by the issue renderer in https://github.com/ansible-community/package-test-results/.

Reply:
> Ah, I see, it's used by the issue renderer in https://github.com/ansible-community/package-test-results/.

Right